controllers/metal3remediation_controller_test.go

```diff
@@ -190,230 +224,373 @@ func setReconcileNormalRemediationExpectations(ctrl *gomock.Controller,
+					objects = append(objects, tc.Metal3Remediation)
 				}
-			})
+				if tc.Machine != nil {
+					objects = append(objects, tc.Machine)
+				}
+				if tc.M3Machine != nil {
+					objects = append(objects, tc.M3Machine)
+				}
+				fakeClient = fake.NewClientBuilder().WithScheme(setupScheme()).WithObjects(objects...).Build()
+			}
+			testReconciler = &Metal3RemediationReconciler{
+				Client:         fakeClient,
+				ManagerFactory: baremetal.NewManagerFactory(fakeClient),
+				Log:            logr.Discard(),
+			}
+			_, err := testReconciler.Reconcile(context.TODO(), tc.TestRequest)
+			evaluateTestError(tc.ExpectedError, err)
+		},
+		Entry("Metal3Remediation haven't been found",
+			reconcileRemediationTestCase{
+				TestRequest:                  defaultTestRequest,
+				ExpectedError:                nil,
+				Metal3RemediationCantBeFound: true,
+			}),
+		Entry("Error retrieving owner machine",
+			reconcileRemediationTestCase{
+				TestRequest:                      defaultTestRequest,
+				ExpectedError:                    &ownerMachineErrorMsg,
+				FailedToCreateRemediationManager: true,
+				Metal3Remediation: &infrav1.Metal3Remediation{ObjectMeta: metav1.ObjectMeta{
+					Name:      metal3RemediationName,
+					Namespace: namespaceName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "Machine",
+							Name:       "wrongName",
+						},
+					},
+				}},
+				Machine: newMachine(clusterName, machineName, metal3machineName, "mynode"),
+			}),
+		Entry("Owner machine not set",
+			reconcileRemediationTestCase{
+				TestRequest:                      defaultTestRequest,
+				ExpectedError:                    &ownerMachineNotSetMsg,
+				FailedToCreateRemediationManager: true,
+				Metal3Remediation: &infrav1.Metal3Remediation{ObjectMeta: metav1.ObjectMeta{
+					Name:      metal3RemediationName,
+					Namespace: namespaceName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "wrongKind",
+							Name:       machineName,
+						},
+					},
+				}},
+				Machine: newMachine(clusterName, machineName, metal3machineName, "mynode"),
+			}),
+		Entry("Failed to retrieve metal3machine",
+			reconcileRemediationTestCase{
+				TestRequest:                      defaultTestRequest,
+				ExpectedError:                    &metal3machineNotFoundMsg,
+				FailedToCreateRemediationManager: true,
+				Metal3Remediation: &infrav1.Metal3Remediation{ObjectMeta: metav1.ObjectMeta{
+					Name:      metal3RemediationName,
+					Namespace: namespaceName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "Machine",
+							Name:       machineName,
+						},
+					},
+				}},
+				Machine:   newMachine(clusterName, machineName, "", "mynode"),
+				M3Machine: newMetal3Machine(metal3machineName, nil, nil, nil, false),
```

Comment:
you are already skipping passing an m3m into the` newMachine()`, but any reason you have to have M3Machine again here considering a test case is for _m3m is not found_?